module-4/backend/week_two.md

```diff
@@ -1,22 +1,46 @@
 ## Week Two - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+1. What's one difference between ES5 and ES6?
+  One difference between ES5 and ES6
+  is the import and export way of requiring files.
 
-Note: When you're done, submit a PR. 
+2. What's the difference between asynchronous and synchronous JavaScript?
+  Asynchronous JavaScript does not wait for a line of code to get executed before running the next one while synchronous JavaScript runs a line at a time.
 
-1. What's one difference between ES5 and ES6?
-2. What's the difference between asynchronous and synchronous JavaScript? 
 3. What are the four pillars of Object Oriented programming?
+  Abstraction, Polymorphism, Inheritance and Encapsulation.
+
 4. What are some tools available in JavaScript to help you write object oriented code?
+  JavaScript has classes and constructor functions to assimilate OOP.
+
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+  Some key factors are the length of the functions and the length of the file. You want to keep everything according to the SRP.
+
+
 6. What's a callback function and what are some reasons when we use/need callback functions?
+  A callback function is a function that is pass as a parameter and it is done so we can keep control on what gets run.
+
 7. What's the scope of variables in Javascript?
+  In JavaScript variables get scoped according to the block they are in.
+
+
 8. What's the difference between `==` and `===` in JavaScript?
+  JavaScript will try to figure out if the the objects being compare can be equal in some way. Meaning that JavaScript will be opinionated while with triple equal JavaScript is not going to be opinionated.
```

Comment:
I know you know this concept but the answer is a bit vague, I'd clarify a little and consider providing an example that demonstrates your understanding!